itests/remove_verifreg_datacap_test.go

```diff
@@ -0,0 +1,347 @@
+	require.NoError(t, err)
+	require.True(t, r.Receipt.ExitCode.IsSuccess())
+
+	ret, err := market.DecodePublishStorageDealsReturn(r.Receipt.Return, build.TestNetworkVersion)
+	require.NoError(t, err)
+	valid, _, err := ret.IsDealValid(0)
+	require.NoError(t, err)
+	require.True(t, valid)
+	dealIds, err := ret.DealIDs()
+	require.NoError(t, err)
+
+	verifiedClientDcap, err := clientApi.StateVerifiedClientStatus(ctx, verifiedClientIDAddr, types.EmptyTSK)
+	require.NoError(t, err)
+
+	// The client has already spent datacap equal to the deal's size -- this will be found in the VerifiedRegistryActor
+	require.Equal(t, big.Sub(datacapToAssign, big.NewIntUnsigned(uint64(dealProposal.PieceSize))), *verifiedClientDcap)
+
+	verifregDcapBefore, err := clientApi.StateVerifiedClientStatus(ctx, builtin.VerifiedRegistryActorAddr, types.EmptyTSK)
+	require.NoError(t, err)
+
+	// Verifreg should now have the datacap for this deal
+	require.Equal(t, big.NewIntUnsigned(uint64(dealProposal.PieceSize)), *verifregDcapBefore)
+
+	// Now let's try to remove datacap from Verifreg
+	removeProposal := verifregst.RemoveDataCapProposal{
+		VerifiedClient:    verifreg.Address,
+		DataCapAmount:     *verifregDcapBefore,
+		RemovalProposalID: verifregst.RmDcProposalID{ProposalID: 0},
+	}
+
+	buf := bytes.Buffer{}
+	buf.WriteString(verifregst.SignatureDomainSeparation_RemoveDataCap)
+	require.NoError(t, removeProposal.MarshalCBOR(&buf), "failed to marshal proposal")
+
+	removeProposalSer := buf.Bytes()
+
+	verifier1Sig, err := clientApi.WalletSign(ctx, verifier1Addr, removeProposalSer)
+	require.NoError(t, err, "failed to sign proposal")
+
+	removeRequest1 := verifregst.RemoveDataCapRequest{
+		Verifier:          verifier1IDAddr,
+		VerifierSignature: *verifier1Sig,
+	}
+
+	verifier2Sig, err := clientApi.WalletSign(ctx, verifier2Addr, removeProposalSer)
+	require.NoError(t, err, "failed to sign proposal")
+
+	removeRequest2 := verifregst.RemoveDataCapRequest{
+		Verifier:          verifier2IDAddr,
+		VerifierSignature: *verifier2Sig,
+	}
+
+	removeDataCapParams := verifregst.RemoveDataCapParams{
+		VerifiedClientToRemove: verifreg.Address,
+		DataCapAmountToRemove:  *verifregDcapBefore,
+		VerifierRequest1:       removeRequest1,
+		VerifierRequest2:       removeRequest2,
+	}
+
+	params, aerr := actors.SerializeParams(&removeDataCapParams)
+	require.NoError(t, aerr)
+
+	m, err = clientApi.MpoolPushMessage(ctx, &types.Message{
+		From:   rootAddr,
+		To:     verifreg.Address,
+		Method: verifreg.Methods.RemoveVerifiedClientDataCap,
+		Params: params,
+		Value:  big.Zero(),
+	}, nil)
+	require.Error(t, err)
+
+	verifregDatacapAfter, err := clientApi.StateVerifiedClientStatus(ctx, builtin.VerifiedRegistryActorAddr, types.EmptyTSK)
+	require.NoError(t, err)
+	require.Equal(t, *verifregDcapBefore, *verifregDatacapAfter) // Verifreg should not have lost datacap
+
+	minerInfo, err := testClient.StateMinerInfo(ctx, testMiner.ActorAddr, types.EmptyTSK)
```

Comment:
Are we gaining much by activating this deal? I'm not sure what assertion we're trying to make.

Reply:
I guess it's a little redundant after checking verifreg's datacap.